pkg/apiserver/middleware_logger.go

```diff
@@ -26,35 +27,53 @@ func LoggingMiddleware(logger log.Logger) gin.HandlerFunc {
 		pathRaw := getPathRaw(ginCtx)
 
 		referer := req.Referer()
+		userAgent := req.UserAgent()
+		status := ginCtx.Writer.Status()
 
-		query := req.URL.Query()
 		queryRaw := req.URL.RawQuery
-		queryParameters := make(map[string]string)
-
-		for k := range query {
-			queryParameters[k] = query.Get(k)
-		}
 
 		method := ginCtx.Request.Method
 		requestTimeNano := time.Since(start)
 		requestTimeSecond := float64(requestTimeNano) / float64(time.Second)
 
 		ctxLogger := chLogger.WithContext(ctx).WithFields(log.Fields{
-			"bytes":                    ginCtx.Writer.Size(),
-			"client_ip":                ginCtx.ClientIP(),
-			"host":                     req.Host,
-			"protocol":                 req.Proto,
-			"request_method":           method,
-			"request_path":             path,
-			"request_path_raw":         pathRaw,
-			"request_query":            queryRaw,
-			"request_query_parameters": queryParameters,
-			"request_referer":          referer,
-			"request_time":             requestTimeSecond,
-			"scheme":                   req.URL.Scheme,
-			"status":                   ginCtx.Writer.Status(),
+			"bytes":              ginCtx.Writer.Size(),
+			"client_ip":          ginCtx.ClientIP(),
+			"host":               req.Host,
+			"protocol":           req.Proto,
+			"request_method":     method,
+			"request_path":       path,
+			"request_path_raw":   pathRaw,
+			"request_query":      queryRaw,
+			"request_referer":    referer,
+			"request_user_agent": userAgent,
+			"request_time":       requestTimeSecond,
+			"scheme":             req.URL.Scheme,
+			"status":             status,
 		})
 
+		// only log query parameters in full for successful requests to avoid logging them from bad crawlers
+		if status != http.StatusUnauthorized && status != http.StatusForbidden && status != http.StatusNotFound {
+			queryParameters := make(map[string]string)
+			query := req.URL.Query()
+
+			for k := range query {
+				queryParameters[k] = query.Get(k)
+			}
+
+			ctxLogger = ctxLogger.WithFields(log.Fields{
+				"request_query_parameters": queryParameters,
+			})
+		}
+
+		if sizeFields := getRequestSizeFields(ctx); sizeFields != nil {
+			ctxLogger = ctxLogger.WithFields(*sizeFields)
+		}
+
+		if sizeFields := getResponseSizeFields(ctx); sizeFields != nil {
+			ctxLogger = ctxLogger.WithFields(*sizeFields)
+		}
+
 		if requestId := req.Header.Get("X-Request-Id"); requestId != "" {
 			ctxLogger = ctxLogger.WithFields(log.Fields{
 				"request_id": requestId,
```

Comment:
would it make sense to do something like:
```go
ctxLoggerFields := log.Fields{
			"bytes":                    ginCtx.Writer.Size(),
			"client_ip":                ginCtx.ClientIP(),
			"host":                     req.Host,
			"protocol":                 req.Proto,
			"request_method":           method,
			"request_path":             path,
			"request_path_raw":         pathRaw,
			"request_query":            queryRaw,
			"request_query_parameters": queryParameters,
			"request_referer":          referer,
			"request_time":             requestTimeSecond,
			"scheme":                   req.URL.Scheme,
			"status":                   ginCtx.Writer.Status(),
			"bytes":              ginCtx.Writer.Size(),
			"client_ip":          ginCtx.ClientIP(),
			"host":               req.Host,
			"protocol":           req.Proto,
			"request_method":     method,
			"request_path":       path,
			"request_path_raw":   pathRaw,
			"request_query":      queryRaw,
			"request_referer":    referer,
			"request_user_agent": userAgent,
			"request_time":       requestTimeSecond,
			"scheme":             req.URL.Scheme,
			"status":             status,
		})

		// only log query parameters in full for successful requests to avoid logging them from bad crawlers
		if status != http.StatusUnauthorized && status != http.StatusForbidden && status != http.StatusNotFound {
			queryParameters := make(map[string]string)
			query := req.URL.Query()

			for k := range query {
				queryParameters[k] = query.Get(k)
			}

			ctxLoggerFields.Add(log.Fields{
				"request_query_parameters": queryParameters,
			}))
		}

		if sizeFields := getRequestSizeFields(ctx); sizeFields != nil {
			ctxLoggerFields.Add(*sizeFields)
		}

		if sizeFields := getResponseSizeFields(ctx); sizeFields != nil {
			ctxLoggerFields.Add(*sizeFields)
		}

		if requestId := req.Header.Get("X-Request-Id"); requestId != "" {
			ctxLogger = ctxLoggerFields.Add(log.Fields{
				"request_id": requestId,
.....
		}
		ctxLogger := chLogger.
					WithContext(ctx).
					WithFields(ctxLoggerFields)
```
?